File: python/rotational-cipher/1/rotational_cipher.py
def rotate(text, key):
    result = ""
    for char in text:
        if 'a' <= char <= 'z':
            cypher = ord(char) + key
            if cypher > ord('z'):
                cypher -= 26
            result += chr(cypher)
        elif 'A' <= char <= 'Z':
            cypher = ord(char) + key
            if cypher > ord('Z'):
                cypher -= 26
            result += chr(cypher)
        else:
            result += char
    return result

File: r/rotational-cipher/1/rotational-cipher.R
auxFunction <- function(x, key){
  res <- x
  if(res > 64 &&  res < 91){
    res <- res + key
    if(res > 90) res <- res-26
  } else if (res > 96 &&  res < 123) {
    res <- res + key
    if(res > 122) res <- res-26
  }
  return(res)
}

rotate <- function(text, key) {
  rotateWithKey <- function(x) auxFunction(x, key)
  utf8ToInt(text) |> sapply(rotateWithKey) |> intToUtf8()
}